codice/ddf
# DDF-3777 Modified IdpEndpoint to respond with a POST response regardless of request binding for Single Sign-On

#### What does this PR do?
Modifies `IdpEndpoint` to respond with a POST response regardless of binding used in the request for Single Sign-On.

#### Who is reviewing it? 
@brjeter @bakejeyner @SmithJosh 

#### Choose 2 committers to review/merge the PR. 
@coyotesqrl 
@stustison

#### How should this be tested? (List steps with links to updated documentation)
1. Full build
2. Install and setup DDF (to test having POST and Redirect bindings in the metadata for SSO and SLO) 
3. To test just having POST Binding in the metadata for SSO: 

- For IdP metadata
  - Go to `https://localhost:8993/services/idp/login/metadata`
  - Copy the XML
  - Delete the `HTTP-Redirect` binding's `SingleSignOnService` line
  - Paste the modified XML in the Admin Console undert `Security -> IdP Client -> Metadata`

- For SP metadata
  - Go to `https://localhost:8993/services/saml/sso/metadata`
  - Copy the XML
  - Delete the `HTTP-Redirect` binding's `AssertionConsumerService` line
  - Paste the modified XML in the Admin Console undert `Security -> IdP Server -> Metadata`

- Logout
- Log back in

4. Test SOAP Binding: 
- Create a CSW source with a username and password
- Verify it's successful
- Query it

5. Test just having Redirect Binding:
- **NOTE: This should be done at the end because it won't let you log back in.**
- Do the above two checklists (under `3`) but instead of deleting the line for `HTTP-Redirect`, delete the one for `HTTP-POST`.
- This shouldn't let you log back in and you should see an `UnsupportedOperationException` with the following message: `HTTP Redirect binding is not supported for single sign-on responses.`

6. It can be further verified by https://github.com/connexta/saml-conformance

#### Any background context you want to provide?
In the Profiles document under section 4.1.2 Profile Overview under step 5. `Identity Provider issues <Response> to Service Provider`:
```
The HTTP Redirect binding MUST NOT be used, as the response will typically exceed the 
URL length permitted by most user agents.
```
#### What are the relevant tickets?
[DDF-3777](https://codice.atlassian.net/browse/DDF-3777)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
